ggml.c

```diff
@@ -11741,6 +11816,9 @@ static thread_ret_t ggml_graph_compute_thread(void * data) {
 }
 
 void ggml_graph_compute(struct ggml_context * ctx, struct ggml_cgraph * cgraph) {
+#ifdef GGML_RECOVERABLE_ERRORS
+    GGML_ASSERT(ctx->last_error_code == GGML_ERRCODE_SUCCESS);
+#endif
```

Comment:
Don't think this is needed

Reply:
The way it works currently is setting an error status in the context when an error occurs. When the error occurs, that operation didn't complete successfully.

Imagine this scenario:

1. Create tensor 1 (succeeds)
2. Create tensor 2 (fails)
3. Run the graph with `ggml_graph_compute`

At step 3, we didn't _actually_ build the graph successfully. We can't compute it, because tensor 2 is missing or in an invalid state. Right?

Now, if it had been tensor 1 that failed, trying to create tensor 2 would have run into the assert and crashed the process: step 3 would never be reached so that's okay.

***

I will make the other changes you suggested so the API stays the same. Can I take this response as indicating you don't have a problem with the general approach I'm using and I should continue to develop it?